h2/src/main/org/h2/command/Parser.java

```diff
@@ -419,6 +423,63 @@
      * The token "!~".
      */
     private static final int NOT_TILDE = COLON_EQ + 1;
+    
+    /**
+     * The PostgreSQL token "->"
+     */
+    private static final int JSON_FIELD = NOT_TILDE + 1;
+    
+    /**
+     * The PostgreSQL token "->>"
+     */
+    private static final int JSON_FIELD_TEXT = JSON_FIELD + 1;
+    
+    /**
+     * The PostgreSQL token "#>"
+     */
+    private static final int JSON_FIELD_PATH = JSON_FIELD_TEXT+ 1;
+    
+    /**
+     * The PostgreSQL token "#>>"
+     */
+    private static final int JSON_FIELD_PATH_TEXT = JSON_FIELD_PATH + 1;
+    
+    /**
+     * The PostgreSQL token "@>"
+     */
+    private static final int JSON_CONTAINS_LEFT = JSON_FIELD_PATH_TEXT + 1;
+    
+    /**
+     * The PostgreSQL token "<@"
+     * Implemented using of "<<@" in case of collision
+     * with SMALLER-AT combination
+     */
+    private static final int JSON_CONTAINS_RIGHT = JSON_CONTAINS_LEFT + 1;
+    
+    /**
+     * The PostgreSQL token "?"
+     */
+    private static final int JSON_EXISTS = JSON_CONTAINS_RIGHT + 1;
+    
+    /**
+     * The PostgreSQL token "?|"
+     */
+    private static final int JSON_EXISTS_ANY = JSON_EXISTS + 1;
+    
+    /**
+     * The PostgreSQL token "?&"
+     */
+    private static final int JSON_EXISTS_ALL = JSON_EXISTS_ANY + 1;
+    
+    /**
+     * The PostgreSQL token "||"
+     */
+    private static final int JSON_CONCAT = JSON_EXISTS_ALL + 1;
```

Comment:
How it is distinguished from a string `||` concatenation and why these constants are not consistent with `TOKENS` array?